libs/platform/src/lib/smart-filter-bar/components/smart-filter-bar-settings-dialog/smart-filter-bar-settings-dialog.component.ts

```diff
@@ -0,0 +1,165 @@
+import { SmartFilterSettingsDialogConfig } from '../../interfaces/smart-filter-bar-settings-dialog-config';
+import { FieldFilterItem } from '../../interfaces/smart-filter-bar-field-filter-item';
+import { SmartFilterBarOptionsDataProvider } from './data-provider';
+
+@Component({
+    selector: 'fdp-smart-filter-bar-settings-dialog',
+    templateUrl: './smart-filter-bar-settings-dialog.component.html',
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [{ provide: RESETTABLE_TOKEN, useExisting: SmartFilterBarSettingsDialogComponent }]
+})
+export class SmartFilterBarSettingsDialogComponent implements Resettable, AfterViewInit, OnDestroy {
+    /**
+     * Table instance
+     */
+    @ViewChild('table') table: Table;
+
+    /**
+     * Table data source.
+     */
+    source: TableDataSource<FieldFilterItem>;
+
+    /** @hidden */
+    _availableFields: FieldFilterItem[];
+
+    /** @hidden */
+    _filterVisibilityOptions: SelectItem[] = [];
+
+    /** @hidden */
+    private _sourceSubscription: Subscription;
+
+    /** @hidden */
+    private _selectedFilters: string[] = [];
+
+    /** An RxJS Subject that will kill the data stream upon component’s destruction (for unsubscribing)  */
+    private readonly _onDestroy$: Subject<void> = new Subject<void>();
+
+    /** @hidden */
+    private _isResetAvailableSubject$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+
+    /** Indicates when reset command is available */
+    readonly isResetAvailable$: Observable<boolean> = this._isResetAvailableSubject$.asObservable();
+
+    /** @hidden */
+    constructor(private _dialogRef: DialogRef<SmartFilterSettingsDialogConfig, string[]>) {
+        this.transformVisibilityLabels();
+        this.setInitialTableState();
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        this.setSelectedFilters();
+    }
+
+    /**
+     * Transforms visibility options into appropriate select item object.
+     */
+    transformVisibilityLabels(): void {
+        for (const [selectValue, selectLabel] of Object.entries(this._dialogRef.data.visibilityCategories)) {
+            this._filterVisibilityOptions.push({
+                label: selectLabel,
+                value: selectValue
+            });
+        }
+    }
+
+    /**
+     * Checks the checkbox in the table row if the field is selected.
+     */
+    setSelectedFilters(): void {
+        this._sourceSubscription?.unsubscribe();
+        this._sourceSubscription = this.source
+            .open()
+            .pipe(takeUntil(this._onDestroy$))
+            .subscribe((items) => {
+                setTimeout(() => {
+                    items.forEach((field, index) => {
+                        if (field.visible) {
+                            this.table.toggleSelectableRow(index);
+                        }
+                    });
+                });
```

Comment:
you can achieve the same behavior with `observeOn`
```suggestion
            .pipe(
            observeOn(asyncScheduler),
            takeUntil(this._onDestroy$)
            )
            .subscribe((items) => {
                    items.forEach((field, index) => {
                        if (field.visible) {
                            this.table.toggleSelectableRow(index);
                        }
                });
```

Reply:
Done